# Continuando con comandos en Terminal desde Linux

2. Comandos Intermedios

wc - Esto nos sirve para contar las lineas, letras, palabras y bytes que tiene el texto o archivo

In [3]:
!wc optd_aircraft_backup.csv

  458  1068 18612 optd_aircraft_backup.csv


esto significa 458 lineas, 1068 palabras y 18612 letras

head - Esto imprime las primeras 10 lineas del texto o archivo, cuando utilizas "head -n K" imprime las primeras K lineas, cuando utilizas "head -c K" imprime los primeros K bytes

In [5]:
!head optd_aircraft_backup.csv

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J


In [6]:
!head -n 20 optd_aircraft_backup.csv

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J
19D,Airbus,A319,,,,,
310,Airbus,A310,,,,,
312,Airbus,A310-200,310,2J,A310,2,J
313,Airbus,A310-300,310,2J,A310,2,J
318,Airbus,A318,32S,2J,A318,2,J
319,Airbus,A319,32S,2J,A319,2,J
31F,Airbus,A310F,,,,,
31W,Airbus,A319 (Sharklets),,,,,
31X,Airbus,A310-200F,31F,2J,A310,2,J
31Y,Airbus,A310-300F,31F,2J,A310,2,J


In [8]:
!head -c 1000 optd_aircraft_backup.csv

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J
19D,Airbus,A319,,,,,
310,Airbus,A310,,,,,
312,Airbus,A310-200,310,2J,A310,2,J
313,Airbus,A310-300,310,2J,A310,2,J
318,Airbus,A318,32S,2J,A318,2,J
319,Airbus,A319,32S,2J,A319,2,J
31F,Airbus,A310F,,,,,
31W,Airbus,A319 (Sharklets),,,,,
31X,Airbus,A310-200F,31F,2J,A310,2,J
31Y,Airbus,A310-300F,31F,2J,A310,2,J
320,Airbus,A320-100/200,32S,2J,A320,2,J
321,Airbus,A321-100/200,32S,2J,A321,2,J
32A,Airbus,A320 (sharklets),32S,2J,A320,2,J
32B,Airbus,A321 (sharklets),32S,2J,A321,2,J
32C,Airbus,A318 (sharklets),,,,,
32D,Airbus,A319 (sharklets),,,,,

tail - Esto imprime las ultimas 10 lineas del texto o archivo, cuando utilizas "head -n K" imprime las ultimas K lineas, cuando utilizas "head -c K" imprime los ultimos K byte

In [9]:
!tail optd_aircraft_backup.csv

TU3,Tupolev,Tu-134,TU3,2J,T134,2,J
TU5,Tupolev,Tu-154,TU5,3J,T154,3,J
VCV,Vickers,Viscount,,,,,
VCX,Vickers,VC-10,,,,,
WWP,IAI,1124 Westwind,WWP,2J,WW24,2,J
YK2,Yakovlev / Jakovlev,Yak 42,YK2,3J,YK42,3,J
YK4,Yakovlev / Jakovlev,Yak 40,YK4,3J,YK40,3,J
YN2,HAMC / Harbin,Y12,YN2,2T,Y12,2,T
YN7,Xian,Yunshuji Y7,YN7,2T,AN24,2,T
YS1,NAMC,YS-11,YS1,2T,YS11,2,T


In [10]:
!tail -n 20 optd_aircraft_backup.csv

SU7,Sukhoi,Superjet 100-75,SU1,2J,ZZZZ,2,J
SU9,Sukhoi,Superjet 100-95,SU1,2J,SU95,2,J
SWF,Fairchild-Swearingen,SA226 Freighter,SWF,2T,*,2,T
SWM,Fairchild-Swearingen,SA26 / SA226 / SA227 Metro / Merlin,SWM,2T,*,2,T
SY8,Shaanxi,Y-8,SY8,4T,AN12,4,T
T20,Tupolev,Tu-204 / Tu-214,T20,2J,T204,2,J
T2F,Tupolev,Tu-204 Freighter,T2F,2J,T204,2,J
T34,Tupolev,Tu-334,T34,2J,T334,2,J
TBM,DAHER-SOCATA,TBM-700,TBM,1T,TBM7,1,T
TRN,Surface Equipment,Train,TRN,S,0000,,S
TU3,Tupolev,Tu-134,TU3,2J,T134,2,J
TU5,Tupolev,Tu-154,TU5,3J,T154,3,J
VCV,Vickers,Viscount,,,,,
VCX,Vickers,VC-10,,,,,
WWP,IAI,1124 Westwind,WWP,2J,WW24,2,J
YK2,Yakovlev / Jakovlev,Yak 42,YK2,3J,YK42,3,J
YK4,Yakovlev / Jakovlev,Yak 40,YK4,3J,YK40,3,J
YN2,HAMC / Harbin,Y12,YN2,2T,Y12,2,T
YN7,Xian,Yunshuji Y7,YN7,2T,AN24,2,T
YS1,NAMC,YS-11,YS1,2T,YS11,2,T


In [11]:
!tail -c 1000 optd_aircraft_backup.csv

D.330,SH3,2T,SH33,2,T
SH6,Shorts,SD.360,SH6,2T,SH36,2,T
SHB,Shorts,SC-5 Belfast,,,,,
SHS,Shorts,SC-7 Skyvan,SHS,2T,SC7,2,T
SSC,Aerospatiale / SNIAS,Concorde,,,,,
SU1,Sukhoi,Superjet 100,,,,,
SU7,Sukhoi,Superjet 100-75,SU1,2J,ZZZZ,2,J
SU9,Sukhoi,Superjet 100-95,SU1,2J,SU95,2,J
SWF,Fairchild-Swearingen,SA226 Freighter,SWF,2T,*,2,T
SWM,Fairchild-Swearingen,SA26 / SA226 / SA227 Metro / Merlin,SWM,2T,*,2,T
SY8,Shaanxi,Y-8,SY8,4T,AN12,4,T
T20,Tupolev,Tu-204 / Tu-214,T20,2J,T204,2,J
T2F,Tupolev,Tu-204 Freighter,T2F,2J,T204,2,J
T34,Tupolev,Tu-334,T34,2J,T334,2,J
TBM,DAHER-SOCATA,TBM-700,TBM,1T,TBM7,1,T
TRN,Surface Equipment,Train,TRN,S,0000,,S
TU3,Tupolev,Tu-134,TU3,2J,T134,2,J
TU5,Tupolev,Tu-154,TU5,3J,T154,3,J
VCV,Vickers,Viscount,,,,,
VCX,Vickers,VC-10,,,,,
WWP,IAI,1124 Westwind,WWP,2J,WW24,2,J
YK2,Yakovlev / Jakovlev,Yak 42,YK2,3J,YK42,3,J
YK4,Yakovlev / Jakovlev,Yak 40,YK4,3J,YK40,3,J
YN2,HAMC / Harbin,Y12,YN2,2T,Y12,2,T
YN7,Xian,Yunshuji Y7,YN7,2T,AN24,2,T
YS1,NAMC,YS-11,YS1,2T,YS11,2,T


find - Esto busca archivos en directorio, utilize "find -type f" para buscar f(archivo),d(directorio). utiliza "find -name nombre" para encontrar por nombre, "find -perm 777" para encontrar archivos con todos los permisos, "find -not nombre" para buscar archivos que no tengan nombre, "find -size KM" para buscar archivos que pesan hasta K megas, "find (ruta)  -exec (ls -la {} \;)" para buscar en la ejecucion ls -la de la ruta (ruta)

In [15]:
!find -type f -name "optd*"

./optd_aircraft_backup.csv
./optd_aircraft.csv


In [58]:
!find ~ -type f -size +100M

/home/dsc/Data/challenge/searches.csv.bz2
/home/dsc/Data/challenge/bookings.csv.bz2
/home/dsc/anaconda3/pkgs/mkl-2018.0.1-h19d6760_4.tar.bz2


In [57]:
!find -perm a+r

./optd_aircraft.csv
./.git/objects/5b/39ea74ad5443c29840ee649cafc3bce0b2cc91
./.git/objects/9f/90bc286a355e394cebd55ad82e81455f5e1bfa
./.git/objects/9f/669cc0ebb32ceaa9ed051d234b77b3386df269
./.git/objects/88/1cd62446ae3ff4fa27a893251f1b8d40bd89ab
./.git/objects/a1/552c8ff26f391c3c6eb02e09c1f372b41f6205
./.git/objects/3b/18e512dba79e4c8300dd08aeb37f8e728b8dad
./.git/objects/ac/4e4b440fc6369e99a27004b565cafdced38bea
./.git/objects/32/a3e157db2aff37cbcc61885f142feadc0d95b3
./.git/objects/b1/ccf13e7cde3012ad7f746d77a6bb25da34de04
./.git/objects/c8/b1517391b7f023d39bb734b10c1c9664dfcd00
./.git/objects/c2/5462e9d16bceaf8c278244b9929541f3768835
./.git/objects/8c/ac0686d5e8a922bef8e5a20ccc63415cffb2ed
./.git/objects/c3/d96464cdd712f1ffdac40082e290ce85cb2fb0
./.git/objects/67/ecf55f952be4195c8f7dfd9902e14fdf78ff80
./.git/objects/6a/b5488185c7770f9d20fd6601501ad9efbdb757
./.git/objects/10/b231e699ac2dbde4796f4dc4225d383b1bba56
./.git/objects/78/cc4be754192cb256c41ccc57b7b7c401003c50
./.git/obje

In [34]:
!find -empty

./.git/branches
./.git/refs/tags
./.git/objects/pack
./.git/objects/info
./copy_directory


In [49]:
!find ~ -type f -size +100M -exec ls -la {} \;

-rwxr-x--- 1 dsc dsc 483188920 mar 13  2018 /home/dsc/Data/challenge/searches.csv.bz2
-rwxr-x--- 1 dsc dsc 554970628 mar 13  2018 /home/dsc/Data/challenge/bookings.csv.bz2
-rw-r--r-- 1 dsc dsc 193625096 feb  6  2018 /home/dsc/anaconda3/pkgs/mkl-2018.0.1-h19d6760_4.tar.bz2


sort - Esto nos sirve para ordenar las lineas de un texto o archivo. utilize "sort -n " para mostrar el orden por valor número, utilize "sort -r " para mostrar el orden reverso,utilize "sort -u " para mostrar el orden unico, utilize "sort -t " " " para ordenar el texto o archivo considerando el delimitador " ",utilize "sort -k M[,N] " para mostrar el orden de las columnas entre M y N

In [7]:
!sort Numbers_Names.txt

1
1
10
11
13
16
2
3
4
5
5
6
6
7
8
9
9
carlos
lucia
miguel
Miguel


In [9]:
!sort -n Numbers_Names.txt

carlos
lucia
miguel
Miguel
1
1
2
3
4
5
5
6
6
7
8
9
9
10
11
13
16


In [10]:
!sort -r Numbers_Names.txt

Miguel
miguel
lucia
carlos
9
9
8
7
6
6
5
5
4
3
2
16
13
11
10
1
1


In [12]:
!sort -u Numbers_Names.txt

1
10
11
13
16
2
3
4
5
6
7
8
9
carlos
lucia
miguel
Miguel


In [13]:
!sort -k 1 Numbers_Names.txt

1
1
10
11
13
16
2
3
4
5
5
6
6
7
8
9
9
carlos
lucia
miguel
Miguel


uniq - Esto nos sirve para mostrar lineas unicas. utilize "uniq -c " para agregar el numero de linea repetido, utilize "uniq -d " para mostrar lineas duplicadas, utilize "uniq -i" para mostrar lineas unicas sin considerar mayusculas y minusculas.

In [16]:
!sort -n Numbers_Names.txt | uniq

carlos
lucia
miguel
Miguel
1
2
3
4
5
6
7
8
9
10
11
13
16


Como vemos aqui estamos usando dos comando sort y uniq, para sacar lineas unicas y ordenadas numericamente

cut - Esto nos sirve para cortar lineas. utilize "cut -d " " " para cortar lineas que tienen delimitador " ", utilize "cut -f " para cortar columnas seleccionadas

In [18]:
!cut -d "," -f 4 optd_aircraft.csv

iata_group
100
146
146
146


14F
14F
14F


En este caso cortamos la columna 4 del archivo optd_aircraft.csv delimitado por comas

paste - Esto nos sirve para pegar o fusionar lineas. utilize "paste -s " para concatenar horizontalmente, utilize "paste -d "," " para concatenar horizontalmente separado por comas, utilize "paste <() <()" para concatenar columnas

In [20]:
!seq 10 | paste

1
2
3
4
5
6
7
8
9
10


In [21]:
!seq 10 | paste -s

1	2	3	4	5	6	7	8	9	10


In [22]:
!seq 10 | paste -s -d ","

1,2,3,4,5,6,7,8,9,10


In [43]:
!paste <(seq 10) <(cat optd_aircraft.csv)

/bin/sh: 1: Syntax error: "(" unexpected


Este codigo pega como columnas, en este ejemplo (seq 10) pega como columna antes de (cat optd_aircraft.csv)

tr - Esto nos sirve para reemplazar caracteres. utilize "tr a A " para reemplazar "a" por "A" en el texto o archivo, utilize "tr -s a A" para reemplazar todos los repetidos "a" por un solo "A", utilize "tr -d a" para borrar caracteres igual a "a", utilize "tr -cd as" para mostrar solo los caracteres igual a "as" en el texto o archivo, utilize tambien los codigos: [:alnum:] todo los digitos y letras [:alpha:] todas las letras,[:blank:]espacios en blanco, [:digit:] todo los digitos, [:lower:] todas las letras minusculas, [:upper:] todas las letras mayusculas

In [48]:
!echo "mmaster  daaaaata   science" | tr -s "[:blank:]" | tr -s "[:alnum:]"

master data science


In [49]:
!echo "master 123 data 1983 science 2" | tr -cd "[:digit:]"

12319832

In [50]:
!echo "master 123 data 1983 science 2" | tr -d "[:digit:]"

master  data  science 


grep - Esto nos sirve para mostrar lineas que siguen un patron. utilize "grep -v a " para mostrar las lineas que no contienen "a", utilize "grep -i a" para mostrar las lineas que contienen "a" sean mayusculas o minusculas, utilize "grep -n a" para mostrar las lineas que contienen "a" agregando el numero de linea, utilize "grep -c a" para mostrar el numero de lineas que contiene "a", utilize "grep -H a" para mostrar el nombre del archivo, utilize "grep -E " " " para mostrar WORKS con expresiones regulares

In [3]:
!grep this Text_example.txt

In [51]:
!grep -v this Text_example.txt

In [52]:
!grep -i this Text_example.txt

In [53]:
!grep -n this Text_example.txt

In [54]:
!grep -c this Text_example.txt

3


In [15]:
!grep -H this Text_example.txt

In [55]:
!grep -E "^T" Text_example.txt

In [58]:
!seq 20 | grep -E "[1-5]{2}"

12


sed - Esto nos sirve para cambiar o reemplazar parte de un texto. utilize comando s para sustituir, despues palabra a sustituir entre "/", despues palabra que sustituira entre "/", utilize g para sustituir todas las palabras, utilize i para sustituir la primera letra, utilize "sed -n 'ap;bp'" para mostrar solamente los digitos "a" y "b".

In [60]:
!echo Sunday | sed s/day/night/

Sunnight


In [61]:
!echo sunday.day.day | sed s/day/night/g

sunnight.night.night


In [62]:
!echo sunday.day.day | sed s/day/night/i

sunnight.day.day


In [70]:
!seq 10 | sed -n '2,4p'

2
3
4


In [72]:
!seq 10 | sed -n '2p;4p'

2
4


Comandos con Archivos comprimidos

las extensiones comunes de archivos son: .zip, .gzm .tar, .tar.gz, bz, bz2

zip, unzip, zipinfo, zcat, zless, zgrep - Estos comandos trabajan con la extension (.zip)

zip - Esto nos sirve para comprimir archivos con extension (.zip)

unzip - Esto nos sirve para descomprimir archivos con extension (.zip)

zipinfo - Esto nos sirve para ver que archivos estan comprimidos e informacion adicional

zcat - Esto nos sirve para ver el contenido del archivo (.zip)

zless - Esto nos sirve para buscar determinado texto en el archivo (.zip)

zgrep - Esto nos sirve para mostrar lineas que siguen un patron en el archivo (.zip)

In [6]:
!zip Files_Comprimidos optd_aircraft_backup.csv Text_example.txt Numbers_Names.txt

  adding: optd_aircraft_backup.csv (deflated 68%)
  adding: Text_example.txt (deflated 30%)
  adding: Numbers_Names.txt (deflated 11%)


In [7]:
ls -la

total 116
drwxr-xr-x 4 dsc dsc  4096 dic 29 00:28 ./
drwxr-xr-x 7 dsc dsc  4096 dic 23 01:38 ../
drwxr-xr-x 2 dsc dsc  4096 dic 23 01:37 copy_directory/
-rw-r--r-- 1 dsc dsc  6627 dic 29 00:28 Files_Comprimidos.zip
drwxr-xr-x 8 dsc dsc  4096 dic 29 00:26 .git/
-rw-r--r-- 1 dsc dsc    12 dic 23 02:05 hello.txt
-rw-r--r-- 1 dsc dsc    65 dic 26 23:35 Numbers_Names.txt
-rw-r--r-- 1 dsc dsc 18612 dic 23 02:01 optd_aircraft_backup.csv
-rwxrwxrwx 1 dsc dsc   460 dic 26 23:17 optd_aircraft.csv*
-rw-r--r-- 1 dsc dsc   790 dic 27 02:56 README.md
-rw-r--r-- 1 dsc dsc 21611 dic 23 02:08 Shell_Basic.ipynb
-rw-r--r-- 1 dsc dsc 27593 dic 27 02:52 Shell_Intermediate.ipynb
-rwxr-xr-x 1 dsc dsc   237 dic 27 01:26 Text_example.txt*


In [10]:
!unzip -l Files_Comprimidos.zip

Archive:  Files_Comprimidos.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
    18612  2018-12-23 02:01   optd_aircraft_backup.csv
      237  2018-12-27 01:26   Text_example.txt
       65  2018-12-26 23:35   Numbers_Names.txt
---------                     -------
    18914                     3 files


In [11]:
!zipinfo Files_Comprimidos.zip

Archive:  Files_Comprimidos.zip
Zip file size: 6627 bytes, number of entries: 3
-rw-r--r--  3.0 unx    18612 tx defN 18-Dec-23 02:01 optd_aircraft_backup.csv
-rwxr-xr-x  3.0 unx      237 tx defN 18-Dec-27 01:26 Text_example.txt
-rw-r--r--  3.0 unx       65 tx defN 18-Dec-26 23:35 Numbers_Names.txt
3 files, 18914 bytes uncompressed, 6107 bytes compressed:  67.7%


In [12]:
!zless Files_Comprimidos.zip

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J
19D,Airbus,A319,,,,,
310,Airbus,A310,,,,,
312,Airbus,A310-200,310,2J,A310,2,J
313,Airbus,A310-300,310,2J,A310,2,J
318,Airbus,A318,32S,2J,A318,2,J
319,Airbus,A319,32S,2J,A319,2,J
31F,Airbus,A310F,,,,,
31W,Airbus,A319 (Sharklets),,,,,
31X,Airbus,A310-200F,31F,2J,A310,2,J
31Y,Airbus,A310-300F,31F,2J,A310,2,J
320,Airbus,A320-100/200,32S,2J,A320,2,J
321,Airbus,A321-100/200,32S,2J,A321,2,J
:5;246mFiles_Comprimidos.zip

In [19]:
!zcat Files_Comprimidos.zip | head

gzip: Files_Comprimidos.zip has more than one entry--rest ignored
iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J


In [27]:
!zgrep -n -H "McDonnell" Files_Comprimidos.zip | head

Files_Comprimidos.zip:250:D10,McDonnell Douglas,DC-10,,,,,
Files_Comprimidos.zip:251:D11,McDonnell Douglas,DC-10-10/15,D10,3J,DC10,3,J
Files_Comprimidos.zip:252:D14,McDonnell Douglas,DC-10-40,,,,,
Files_Comprimidos.zip:253:D1C,McDonnell Douglas,DC-10-30/40,D10,3J,DC10,3,J
Files_Comprimidos.zip:254:D1F,McDonnell Douglas,DC-10 Freighter,,,,,
Files_Comprimidos.zip:255:D1M,McDonnell Douglas,DC-10 Combi,D1M,3J,DC10,3,J
Files_Comprimidos.zip:256:D1X,McDonnell Douglas,DC-10-10 Freighter,D1F,3J,DC10,3,J
Files_Comprimidos.zip:257:D1Y,McDonnell Douglas,DC-10-30 / 40 Freighter,D1F,3J,DC10,3,J
Files_Comprimidos.zip:393:M11,McDonnell Douglas,MD-11,M11,3J,MD11,3,J
Files_Comprimidos.zip:394:M1F,McDonnell Douglas,MD-11 Freighter,M1F,3J,MD11,3,J


In [28]:
!unzip -c Files_Comprimidos.zip Numbers_Names.txt | head

Archive:  Files_Comprimidos.zip
  inflating: Numbers_Names.txt       
1
2
3
4
5
6
7
8


In [29]:
!unzip -p Files_Comprimidos.zip Numbers_Names.txt | head

1
2
3
4
5
6
7
8
9
10


gzip, zunzip, zcat, zless, zgrep - Estos comandos trabajan con la extension (.gz)

gzip - Esto nos sirve para comprimir archivos con extension (.gz), utilize "gzip -d " para descomprimir

gunzip - Esto nos sirve para descomprimir archivos con extension (.gz)

zcat - Esto nos sirve para ver el contenido del archivo (.gz)

zless - Esto nos sirve para buscar determinado texto en el archivo (.gz)

zgrep - Esto nos sirve para mostrar lineas que siguen un patron en el archivo (.gz)

In [23]:
!gzip optd_aircraft_backup.csv

In [24]:
ls -la

total 104
drwxr-xr-x 4 dsc dsc  4096 dic 29 00:50 ./
drwxr-xr-x 7 dsc dsc  4096 dic 23 01:38 ../
drwxr-xr-x 2 dsc dsc  4096 dic 23 01:37 copy_directory/
-rw-r--r-- 1 dsc dsc  6627 dic 29 00:28 Files_Comprimidos.zip
drwxr-xr-x 8 dsc dsc  4096 dic 29 00:36 .git/
-rw-r--r-- 1 dsc dsc    12 dic 23 02:05 hello.txt
-rw-r--r-- 1 dsc dsc    65 dic 26 23:35 Numbers_Names.txt
-rw-r--r-- 1 dsc dsc  5927 dic 23 02:01 optd_aircraft_backup.csv.gz
-rwxrwxrwx 1 dsc dsc   460 dic 26 23:17 optd_aircraft.csv*
-rw-r--r-- 1 dsc dsc   790 dic 27 02:56 README.md
-rw-r--r-- 1 dsc dsc 21611 dic 23 02:08 Shell_Basic.ipynb
-rw-r--r-- 1 dsc dsc 27593 dic 27 02:52 Shell_Intermediate.ipynb
-rwxr-xr-x 1 dsc dsc   237 dic 27 01:26 Text_example.txt*


In [26]:
!gzip -l optd_aircraft_backup.csv.gz

         compressed        uncompressed  ratio uncompressed_name
               5927               18612  68.4% optd_aircraft_backup.csv


In [31]:
!gzip optd_aircraft_backup.csv Text_example.txt Numbers_Names.txt

gzip: optd_aircraft_backup.csv: No such file or directory
gzip: Text_example.txt: No such file or directory
gzip: Numbers_Names.txt: No such file or directory


bzip2, bunzip2, bzcat, bzless, bzgrep - Estos comandos trabajan con la extension (.bz2)

bzip2 - Esto nos sirve para comprimir archivos hacia extension (.bz2), utilize "bzip2 -d " para descomprimir archivos con extencion (.bz2)

In [40]:
!bzip2 -k optd_aircraft_backup.csv

In [41]:
ls -la

total 124
drwxr-xr-x 4 dsc dsc  4096 dic 29 00:58 ./
drwxr-xr-x 7 dsc dsc  4096 dic 23 01:38 ../
drwxr-xr-x 2 dsc dsc  4096 dic 23 01:37 copy_directory/
-rw-r--r-- 1 dsc dsc  6627 dic 29 00:28 Files_Comprimidos.zip
drwxr-xr-x 8 dsc dsc  4096 dic 29 00:36 .git/
-rw-r--r-- 1 dsc dsc    12 dic 23 02:05 hello.txt
-rw-r--r-- 1 dsc dsc    65 dic 26 23:35 Numbers_Names.txt
-rw-r--r-- 1 dsc dsc 18612 dic 23 02:01 optd_aircraft_backup.csv
-rw-r--r-- 1 dsc dsc  5900 dic 23 02:01 optd_aircraft_backup.csv.bz2
-rwxrwxrwx 1 dsc dsc   460 dic 26 23:17 optd_aircraft.csv*
-rw-r--r-- 1 dsc dsc   790 dic 27 02:56 README.md
-rw-r--r-- 1 dsc dsc 21611 dic 23 02:08 Shell_Basic.ipynb
-rw-r--r-- 1 dsc dsc 27593 dic 27 02:52 Shell_Intermediate.ipynb
-rwxr-xr-x 1 dsc dsc   237 dic 27 01:26 Text_example.txt*


tar - Estos comandos trabajan con la extension (.tar)
-c = crear;
-r = agregar;
-x = extracto;
-t = lista / vista;
-f = archivo archivado;
-v: verbosa;
-z = zip;
-j = bzip2

Comandos con archivos csv

csvlook - Esto renderiza un archivo CSV en la consola como una tabla de ancho fijo.
-d = delimitador;
-H = el archivo csv no tiene fila de encabezado;
-l = mostrar números de línea

csvstat - Esto Imprime estadísticas descriptivas para cada columna en un archivo CSV.
-d = delimitador;
-H = el archivo csv no tiene fila de encabezado;
-l = mostrar números de línea

In [47]:
!csvlook optd_aircraft.csv | head

| iata_code | manufacturer | model                               | iata_group | iata_category | icao_code | nb_engines | aircraft_type |
| --------- | ------------ | ----------------------------------- | ---------- | ------------- | --------- | ---------- | ------------- |
| 100       | Fokker       | 100                                 | 100        | 2J            | F100      |          2 | J             |
| 141       | BAe          | 146-100                             | 146        | 4J            | B461      |          4 | J             |
| 142       | BAe          | BAE Systems 146-200 Passenger       | 146        | 4J            | B462      |          4 | J             |
| 143       | BAe          | 146-300                             | 146        | 4J            | B463      |          4 | J             |
| 146       | BAe          | 146                                 |            |               |           |            |               |
| 14F       | BAe          | 146 Freighte

In [48]:
!csvlook -d "," optd_aircraft.csv | head

| iata_code | manufacturer | model                               | iata_group | iata_category | icao_code | nb_engines | aircraft_type |
| --------- | ------------ | ----------------------------------- | ---------- | ------------- | --------- | ---------- | ------------- |
| 100       | Fokker       | 100                                 | 100        | 2J            | F100      |          2 | J             |
| 141       | BAe          | 146-100                             | 146        | 4J            | B461      |          4 | J             |
| 142       | BAe          | BAE Systems 146-200 Passenger       | 146        | 4J            | B462      |          4 | J             |
| 143       | BAe          | 146-300                             | 146        | 4J            | B463      |          4 | J             |
| 146       | BAe          | 146                                 |            |               |           |            |               |
| 14F       | BAe          | 146 Freighte

In [53]:
!csvlook -ld "," optd_aircraft.csv | head

| line_numbers | iata_code | manufacturer | model                               | iata_group | iata_category | icao_code | nb_engines | aircraft_type |
| ------------ | --------- | ------------ | ----------------------------------- | ---------- | ------------- | --------- | ---------- | ------------- |
|            1 | 100       | Fokker       | 100                                 | 100        | 2J            | F100      |          2 | J             |
|            2 | 141       | BAe          | 146-100                             | 146        | 4J            | B461      |          4 | J             |
|            3 | 142       | BAe          | BAE Systems 146-200 Passenger       | 146        | 4J            | B462      |          4 | J             |
|            4 | 143       | BAe          | 146-300                             | 146        | 4J            | B463      |          4 | J             |
|            5 | 146       | BAe          | 146                                 |       

In [51]:
!csvstat -d "," optd_aircraft.csv | head

  1. "iata_code"

	Type of data:          Text
	Contains null values:  False
	Unique values:         9
	Longest value:         3 characters
	Most common values:    100 (1x)
	                       141 (1x)
	                       142 (1x)
	                       143 (1x)


In [54]:
!csvstat -d "," -c 2,4 optd_aircraft.csv | head

  2. "manufacturer"

	Type of data:          Text
	Contains null values:  False
	Unique values:         2
	Longest value:         6 characters
	Most common values:    BAe (8x)
	                       Fokker (1x)

  4. "iata_group"


csvcut - Esto nos sirve para filtrar y truncar archivos CSV.
-c = columna;
-n = Mostrar nombres de columna e índices

In [56]:
!csvcut -n optd_aircraft.csv

  1: iata_code
  2: manufacturer
  3: model
  4: iata_group
  5: iata_category
  6: icao_code
  7: nb_engines
  8: aircraft_type


In [59]:
!csvcut -d "," -c 2,3,4 optd_aircraft.csv

manufacturer,model,iata_group
Fokker,100,100
BAe,146-100,146
BAe,BAE Systems 146-200 Passenger,146
BAe,146-300,146
BAe,146,
BAe,146 Freighter (-100/200/300QT & QC),
BAe,146 Freighter (-100QT & QC),14F
BAe,146 Freighter (-200QT & QC),14F
BAe,146 Freighter (-300QT & QC),14F


csvgrep - Buscar archivos CSV.
-m = patrón;
-i = invertir el resultado;
-a = cualquier columna listada debe coincidir con la cadena de búsqueda (por defecto es todo)

In [64]:
!csvgrep -d "," -c iata_code -m 380 optd_aircraft_backup.csv

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
380,Airbus,A380,,,,,


In [66]:
!csvgrep -d "," -c iata_code -im 380 optd_aircraft_backup.csv | head

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
100,Fokker,100,100,2J,F100,2,J
141,BAe,146-100,146,4J,B461,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
143,BAe,146-300,146,4J,B463,4,J
146,BAe,146,,,,,
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J


In [67]:
!csvgrep -d "," -a -c 1,2,3 -m 380 optd_aircraft_backup.csv | head

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
380,Airbus,A380,,,,,
388,Airbus,A380,380,4J,A388,4,J
38F,Airbus,A380F,38F,4J,A388,4,J


csvsort - Ordenar archivos CSV.
-r = marcha atrás;
-n = Mostrar nombres de columna e índices

In [68]:
!csvsort -n -d "," optd_aircraft.csv

  1: iata_code
  2: manufacturer
  3: model
  4: iata_group
  5: iata_category
  6: icao_code
  7: nb_engines
  8: aircraft_type


In [73]:
!csvsort -d "," -c 1 -r optd_aircraft.csv

iata_code,manufacturer,model,iata_group,iata_category,icao_code,nb_engines,aircraft_type
14Z,BAe,146 Freighter (-300QT & QC),14F,4J,B463,4,J
14Y,BAe,146 Freighter (-200QT & QC),14F,4J,B462,4,J
14X,BAe,146 Freighter (-100QT & QC),14F,4J,B461,4,J
14F,BAe,146 Freighter (-100/200/300QT & QC),,,,,
146,BAe,146,,,,,
143,BAe,146-300,146,4J,B463,4,J
142,BAe,BAE Systems 146-200 Passenger,146,4J,B462,4,J
141,BAe,146-100,146,4J,B461,4,J
100,Fokker,100,100,2J,F100,2,J


csvformat - Convierte un archivo CSV a un formato de salida personalizado.
-D = delimitador de salida

In [74]:
!csvformat -d "," -D "~" ./optd_aircraft.csv > ./optd_aircraft_new.csv

In [76]:
ls -la

total 128
drwxr-xr-x 4 dsc dsc  4096 dic 29 01:31 ./
drwxr-xr-x 7 dsc dsc  4096 dic 23 01:38 ../
drwxr-xr-x 2 dsc dsc  4096 dic 23 01:37 copy_directory/
-rw-r--r-- 1 dsc dsc  6627 dic 29 00:28 Files_Comprimidos.zip
drwxr-xr-x 8 dsc dsc  4096 dic 29 01:31 .git/
-rw-r--r-- 1 dsc dsc    12 dic 23 02:05 hello.txt
-rw-r--r-- 1 dsc dsc    65 dic 26 23:35 Numbers_Names.txt
-rw-r--r-- 1 dsc dsc 18612 dic 23 02:01 optd_aircraft_backup.csv
-rw-r--r-- 1 dsc dsc  5900 dic 23 02:01 optd_aircraft_backup.csv.bz2
-rwxrwxrwx 1 dsc dsc   460 dic 26 23:17 optd_aircraft.csv*
-rw-r--r-- 1 dsc dsc   460 dic 29 01:31 optd_aircraft_new.csv
-rw-r--r-- 1 dsc dsc   790 dic 27 02:56 README.md
-rw-r--r-- 1 dsc dsc 21611 dic 23 02:08 Shell_Basic.ipynb
-rw-r--r-- 1 dsc dsc 27593 dic 27 02:52 Shell_Intermediate.ipynb
-rwxr-xr-x 1 dsc dsc   237 dic 27 01:26 Text_example.txt*
